consensus/spos/bls/subroundEndRound.go

```diff
@@ -281,6 +305,70 @@ func (sr *subroundEndRound) doEndRoundJobByLeader() bool {
 	return true
 }
 
+// TODO:
+//	- slashing for invalid sig shares
+//	- handle sig share verifications concurrently
+func (sr *subroundEndRound) verifyNodesOnAggSigVerificationFail() error {
+	multiSigner := sr.MultiSigner()
+	pubKeys := sr.ConsensusGroup()
+
+	for i, pk := range pubKeys {
+		isJobDone, err := sr.JobDone(pk, SrSignature)
+		if err != nil || !isJobDone {
+			continue
+		}
+
+		sigShare, err := multiSigner.SignatureShare(uint16(i))
+		if err != nil {
+			return err
+		}
+
+		isSuccessfull := true
+		err = multiSigner.VerifySignatureShare(uint16(i), sigShare, sr.GetData(), nil)
+		if err != nil {
+			isSuccessfull = false
+
+			err = sr.SetJobDone(pk, SrSignature, false)
```

Comment:
also the sr.PeerHonestyHandler() should be decreased? maybe twice? It was increased once on the receivedSignature and now it was found it is invalid.

Reply:
:+1: will include in the next PR when slashing will be applied

Reply:
as discussed, decreased peerHonesty here to be in line with the old setup, since the peerHonesty has been increased optimistically for the malicious signer